File: modules/wake-stop-module/ios/WakeStopWakeModule.swift
import ExpoModulesCore

public class WakeStopWakeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("WakeStopWakeModule")

    AsyncFunction("registerWake") { (payload: String) async throws -> [String: Any] in
      // TODO: Parse trip payload and register Core Location geofence + notification fallback.
      return [
        "geofenceArmed": false,
        "fallbackAlarmArmed": false,
        "nativeRegistrationSucceeded": false,
        "reason": "iOS native wake flow not wired yet."
      ]
    }

    AsyncFunction("cancelWake") { (tripId: String) async in
      _ = tripId
      // TODO: Stop region monitoring and pending notifications for trip.
    }

    AsyncFunction("triggerTestAlarm") {
      // TODO: Trigger local test notification/haptic sequence.
    }
  }
}

File: modules/wake-stop-module/android/src/main/java/expo/modules/wakestop/WakeStopWakeModule.kt
package expo.modules.wakestop

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class WakeStopWakeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("WakeStopWakeModule")

    AsyncFunction("registerWake") { payload: String ->
      // TODO: Parse payload and register Android geofence + alarm fallback.
      mapOf(
        "geofenceArmed" to false,
        "fallbackAlarmArmed" to false,
        "nativeRegistrationSucceeded" to false,
        "reason" to "Android native wake flow not wired yet."
      )
    }

    AsyncFunction("cancelWake") { tripId: String ->
      // TODO: Cancel geofence/alarm for trip.
      tripId
    }

    AsyncFunction("triggerTestAlarm") {
      // TODO: Trigger local test alert flow.
    }
  }
}
